x/group/client/testutil/tx.go

```diff
@@ -2303,6 +2303,286 @@ func (s *IntegrationTestSuite) TestTxLeaveGroup() {
+	  },{
+		"address": "%s",
+		  "weight": "1",
+		  "metadata": "AQ=="
+	  },{
+		"address": "%s",
+		  "weight": "3",
+		  "metadata": "AQ=="
+	  }]}`, members[0], members[1], members[2])
+	validMembersFile := testutil.WriteToNewTempFile(s.T(), validMembers)
+	out, err := cli.ExecTestCLICmd(clientCtx, client.MsgCreateGroupCmd(),
+		append([]string{
+			val.Address.String(),
+			validMetadata,
+			validMembersFile.Name(),
+		},
+			commonFlags...,
+		),
+	)
+	require.NoError(err, out.String())
+	var txtResp sdk.TxResponse
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txtResp), out.String())
+	groupID := s.getGroupIdFromTxResponse(txtResp)
+
+	// create group policy
+	out, err = cli.ExecTestCLICmd(clientCtx, client.MsgCreateGroupPolicyCmd(),
+		append(
+			[]string{
+				val.Address.String(),
+				groupID,
+				"AQ==",
+				"{\"@type\":\"/cosmos.group.v1.ThresholdDecisionPolicy\", \"threshold\":\"4\", \"windows\":{\"voting_period\":\"1s\"}}",
+			},
+			commonFlags...,
+		),
+	)
+	var txResp = sdk.TxResponse{}
+	require.NoError(err, out.String())
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txResp), out.String())
+	s.Require().Equal(uint32(0), txResp.Code, out.String())
+
+	out, err = cli.ExecTestCLICmd(val.ClientCtx, client.QueryGroupPoliciesByGroupCmd(), []string{groupID, fmt.Sprintf("--%s=json", tmcli.OutputFlag)})
+	s.Require().NoError(err, out.String())
+	var res group.QueryGroupPoliciesByGroupResponse
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &res))
+	groupPolicyAdress := res.GroupPolicies[0].Address
+
+	testCases := []struct {
+		name      string
+		args      []string
+		expectErr bool
+		errMsg    string
+		respType  proto.Message
+	}{
+		{
+			"member that leaves does not affect the threshold",
+			append(
+				[]string{
+					members[0],
+					groupID,
+					fmt.Sprintf("--%s=%s", flags.FlagFrom, members[2]),
+				},
+				commonFlags...,
+			),
+			false,
+			"",
+			&sdk.TxResponse{},
+		},
+		{
+			"member that leaves affects the threshold",
+			append(
+				[]string{
+					members[2],
+					groupID,
+					fmt.Sprintf("--%s=%s", flags.FlagFrom, members[2]),
+				},
+				commonFlags...,
+			),
+			true,
+			"failed to execute message",
+			&sdk.TxResponse{},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmdSubmitProposal := client.MsgSubmitProposalCmd()
+			cmdLeaveGroup := client.MsgLeaveGroupCmd()
+
+			out, err := cli.ExecTestCLICmd(clientCtx, cmdLeaveGroup, tc.args)
+			require.NoError(err, out.String())
+			var resp sdk.TxResponse
+			require.NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &resp), out.String())
+
+			submitPropossalArgs := append([]string{
+				s.createCLIProposal(
+					groupPolicyAdress, val.Address.String(),
+					groupPolicyAdress, val.Address.String(),
+					"",
+				),
+			},
+				commonFlags...,
+			)
+			out, err = cli.ExecTestCLICmd(clientCtx, cmdSubmitProposal, submitPropossalArgs)
```

Comment:
Here we're testing:
- member leaves group
- submit proposal
- no error

A more useful test would be IMO:
1. submit proposal
2. all 3 members vote yes on this proposal
3. then, member leaves group (table test for which member leaves)
4. wait until voting period end
5. depending on which member left, the proposal's status should be ACCEPTED or REJECTED